src/components/MobileWalletDialog/routes/RenameWallet.tsx

```diff
@@ -0,0 +1,98 @@
+import { Button, FormControl, FormErrorMessage, Input } from '@chakra-ui/react'
+import { useCallback, useState } from 'react'
+import { useTranslate } from 'react-polyglot'
+import { useHistory, useLocation } from 'react-router'
+import { DialogBackButton } from 'components/Modal/components/DialogBackButton'
+import { DialogBody } from 'components/Modal/components/DialogBody'
+import { DialogFooter } from 'components/Modal/components/DialogFooter'
+import {
+  DialogHeader,
+  DialogHeaderLeft,
+  DialogHeaderMiddle,
+} from 'components/Modal/components/DialogHeader'
+import { DialogTitle } from 'components/Modal/components/DialogTitle'
+import { SlideTransition } from 'components/SlideTransition'
+import { Text } from 'components/Text'
+import { updateWallet } from 'context/WalletProvider/MobileWallet/mobileMessageHandlers'
+import type { MobileLocationState } from 'context/WalletProvider/MobileWallet/types'
+
+import { MobileWalletDialogRoutes } from '../types'
+
+const isValidLabel = (label: unknown): label is string => {
+  return typeof label === 'string' && label.length > 0 && label.length < 65
+}
+
```

Comment:
preferably blocking: there are 2 concepts mixed into one typeguard, validating the type: this is defined and a string, and validating the length which should be a separate assert like `assertLabelLength`, both functions could be inside an util file

Reply:
Think the util is fine, though I don't really see a point in `typeof` checks here, can simply do `label && label.length < 65`.
A better option is to early return `if (!label)`on l.34 here:
https://github.com/shapeshift/web/blob/2c09f0e47c85c987ccd47d19cd3f6a1d033d5f64/src/components/MobileWalletDialog/routes/RenameWallet.tsx#L33-L37

That way the only check of this method will be `label.length < 65` (0-length string is falsy and already handled by the check proposed above).
Could also do with a more descriptive name (`isValidLabelLength`) and proper arg type (`string | null` currently, or `string` if using the suggestion above)